bench/src/main/scala/cats/parse/bench/self.scala

```diff
@@ -47,22 +47,24 @@ object Json {
       (whitespaces0.with1.soft ~ P.char(',') ~ whitespaces0).void
 
     def rep[A](pa: P1[A]): P[List[A]] =
-      (whitespaces0 *> P.repSep(pa, min = 0, sep = listSep) <* whitespaces0)
+      P.repSep(pa, min = 0, sep = listSep).surroundedBy(whitespaces0)
 
-    val list = (P.char('[') *> rep(recurse) <* P.char(']'))
+    val list = rep(recurse).with1
+      .between(P.char('['), P.char(']'))
       .map { vs => JArray.fromSeq(vs) }
 
     val kv: P1[(String, JValue)] =
       justStr ~ ((whitespaces0.with1 ~ P.char(':') ~ whitespaces0) *> recurse)
```

Comment:
This can use `P.char(':').surroundedBy(whitespace0)`

Reply:
ah yes!